_knitr/A Shiny app for Hong Kong District Councillors_20200801.md

```diff
@@ -8,43 +8,160 @@ output:                    # DO NOT CHANGE
+
+## 🗄️ How was the data collected?
+
+Since there was no existing single data source on the DCs, we had to put this togther ourselves. All the data on each District Councillor, their constituency, the party they belong to, and their Facebook page was all collected manually through a combination of Wikipedia and Facebook. The data was initially housed on Google Sheets, for multiple reasons: 
+
+1. Using Google Sheets made it easy for multiple people to collaborate on data entry.
+2. Keeping the data outside of the repo has the advantage of keeping the memory size minimal, in line with best practices.
+3. By storing the data in Google Sheets, non-technical users would also be able to access the data too.
+
+Most of all, it was plain easy to access the Google Sheets data with the {googlesheets4} package! A key function is `googlesheets4::gs4_auth()`, which directs the developer to a web browser, asked to sign in to their Google account, and to grant googlesheets4 permission to operate on their behalf with Google Sheets. This Google Sheet is available [here](https://docs.google.com/spreadsheets/d/1007RLMHSukSJ5OfCcDJdnJW5QMZyS2P-81fe7utCZwk/).
+
+<img src="https://raw.githubusercontent.com/martinctc/blog/master/images/googlesheet-example.png" width=500>
+
+Creating a map with constituency boundaries also required additional data. Boundaries for each constituency were obtained through a Freedom of Information (FOI) request by a member of the public [here](https://accessinfo.hk/en/request/shapefileshp_for_2019_district_c).
+
+This was pretty much Phase #1 of data collection, where we had single Google Sheet with basic information about the District Councillors and their Facebook feeds, which enabled us to create a proof of concept of the Shiny app, i.e. making sure that we can set up a mechanism where the user can select a constituency and the app returns the corresponding Facebook feed of the District Councillor.
+
+Based on user feedback, we started with Phase #2 of data collection, which involved a web-scraping exercise on the official [Hong Kong District Council website](https://www.districtcouncils.gov.hk/index.html) and the [HK01 News Page on the 2019 District Council elections](https://dce2019.hk01.com/) to get extra data points, such as:
+- Contact email address
+- Contact number
+- Office address
+- Number of votes, and share of votes won in 2019
 
-## A solution for getting the Facebook feed
+A function that was extremely helpful for figuring out the URL of the District Councillors' individual official pages is the following. What this does is to run a Bing search on the https://www.districtcouncils.gov.hk website, and scrape from the search result any links which match what we want (based on what the URL string looks like). Although this doesn't always work, it helped us a long way with the 452 District Councillors.
```

Comment:
Is it worth saying why you had to use Bing search instead of Google, Yahoo, Ask Jeeves etc.?